tests/test_sampler.py

```diff
@@ -417,31 +450,36 @@ def test_parse_sampling_strategy():
             ]
         }
     }
-    """
+
     # noinspection PyProtectedMember
-    s, strategies = sampler._parse_sampling_strategy(None, response)
-    assert '%s' % s == 'AdaptiveSampler(0.001, 2, 10)'
-    assert strategies
+    remote_sampler._update_sampler(response)
+    assert '%s' % remote_sampler.sampler == 'AdaptiveSampler(0.001, 2, 10)'
 
-    existing_strategies = {
-        "defaultSamplingProbability":0.51,
-        "defaultLowerBoundTracesPerSecond":3,
-        "perOperationStrategies":
+    new_response = {
+        "strategyType":1,
+        "operationSampling":
+        {
+            "defaultSamplingProbability":0.51,
+            "defaultLowerBoundTracesPerSecond":3,
+            "perOperationStrategies":
             [
                 {
                     "operation":"op",
                     "probabilisticSampling":{
-                        "samplingRate":0.5
+                        "samplingRate":0.002
                     }
                 }
             ]
+        }
     }
-    existing_sampler = AdaptiveSampler(existing_strategies, 2)
-    assert '%s' % existing_sampler == 'AdaptiveSampler(0.51, 3, 2)'
 
     # noinspection PyProtectedMember
-    s, strategies = sampler._parse_sampling_strategy(existing_sampler, response)
-    assert '%s' % existing_sampler == 'AdaptiveSampler(0.51, 3, 2)'
-    assert strategies
+    remote_sampler._update_sampler(new_response)
+    assert '%s' % remote_sampler.sampler == 'AdaptiveSampler(0.51, 3, 10)'
 
-    sampler.close()
+    # noinspection PyProtectedMember
+    remote_sampler._update_sampler({"strategyType":0,"probabilisticSampling":{"samplingRate":0.004}})
+    assert '%s' % remote_sampler.sampler == 'ProbabilisticSampler(0.004)', \
+        'should not fail going from adaptive sampler to probabilistic sampler'
+
+    remote_sampler.close()
```

Comment:
nit: the last two tests are almost 100 lines each. Perhaps there's a simpler way to structure them? Can they be data driven?